feat(docs): New docs from i18n markdown files

benhalverson opened the pull request:
<!--
Please read the [Code of Conduct](https://github.com/nodejs/nodejs.dev/blob/master/CODE_OF_CONDUCT.md) and the [Contributing Guidelines](https://github.com/nodejs/nodejs.dev/blob/master/CONTRIBUTING.md) before opening a pull request.
-->

## Description

WIP Do not merge

## Related Issues

<!--
  Link to the issue that is fixed by this PR (if there is one)
  e.g. Fixes #1234, Addresses #1234, Related to #1234, etc.
-->

<!--
  If you want to generate a preview of this PR on our staging server please
  make a comment on the Pull-Request with the text `/preview`
 -->

zeke commented:
Exciting! Let me know if I can help.

benhalverson commented:
Is it possible to add frontmatter to the i18n markdown files? @zeke 
something like 

Instead of the below from https://raw.githubusercontent.com/nodejs/i18n/master/content/v5.x/en-US/doc/api/buffer.md
```
# Buffer

    Stability: 2 - Stable

Prior to the introduction of `TypedArray` in ECMAScript 2015 (ES6), the
JavaScript language had no mechanism for reading or manipulating streams
of binary data. The `Buffer` class was introduced as part of the Node.js
API to make it possible to interact with octet streams in the context of things
like TCP streams and file system operations.

```

It looks like this.

```
---
title: Buffer
url: buffer
stability: 2
---

# Buffer

Prior to the introduction of `TypedArray` in ECMAScript 2015 (ES6), the
JavaScript language had no mechanism for reading or manipulating streams
of binary data. The `Buffer` class was introduced as part of the Node.js
API to make it possible to interact with octet streams in the context of things
like TCP streams and file system operations.

```

nschonni commented:
@benhalverson those files are just directly copied from the nodejs/node repo https://github.com/nodejs/i18n/blob/master/.github/workflows/nightly-sync.yml

zeke commented:
Like @nschonni said, there are not currently any transformations to the markdown files. They are left in the original format, and I think we should probably keep it that way.

What hurdle are you hitting? What about creating a function here that could take a markdown string as input and extract the metadata from it?

benhalverson commented:
The issue that I am having is making a proper Graphql query that gives me a single document with data that I can use to display on the screen.

The common way to use Graphql and markdown files with Gatsby is to have frontmatter for example.  

A typical query https://github.com/resir014/gatsby-starter-typescript-plus/blob/master/src/templates/page.tsx
```
  query PageTemplateQuery($slug: String!) {
    site {
      siteMetadata {
        title
        description
        author {
          name
          url
        }
      }
    }
    markdownRemark(fields: { slug: { eq: $slug } }) {
      html
      excerpt
      frontmatter {
        title
      }
    }
  }
```
I was trying to keep with the best practices/ common patterns. I can keep digging. 
For the existing metadata, is it anything inside `<!-- -->` like
 `<!--introduced_in=v0.1.90-->` or
```
<!-- YAML
changes:
  - version: v6.4.0
    pr-url: https://github.com/nodejs/node/pull/7111
    description: Introduced `latin1` as an alias for `binary`.
  - version: v5.0.0
    pr-url: https://github.com/nodejs/node/pull/2859
    description: Removed the deprecated `raw` and `raws` encodings.
-->
```

zeke commented:
You also mentioned these:

```
---
title: Buffer
url: buffer
stability: 2
```

...which are not in HTML comments.

I haven't used Gatsby before. Is there a way to preprocess the content before querying it?

jemjam commented:
Some discussion today during our meeting about adding front matter to more of the root documents: possibly get some folk from i18n also discussing this though.
